src/drivers/hx711.rs

```diff
@@ -0,0 +1,201 @@
+//! HX711 driver - 24-bit Analog-to-Digital Converter (ADC).
+//!
+//! ## Note
+//!
+//! This driver, is still under progress...
+#![no_main]
+#![no_std]
+#![allow(unused)]
+#![allow(missing_docs)]
+
+use cortex_m::asm as _;
+use core::future::Future;
+use cortex_m_rt::entry;
+use defmt::{unwrap, Format};
+use defmt_rtt as _;
+use embedded_hal::{
+    delay::DelayNs,
+    digital::{InputPin, OutputPin},
+};
+use rtic_monotonics::{
+    nrf::timer::{ExtU64, Timer0 as Mono},
+    systick::fugit::Duration,
+    Monotonic,
+};
+
+pub type Instant = <Mono as rtic_monotonics::Monotonic>::Instant;
+
+
+/// Hx711 driver struct.
+pub struct Hx711Driver<PINOUT: OutputPin + Send, PININ: InputPin + Send> {
+    pub pd_sck: PINOUT,
+    pub dout: PININ,
+    pub dout_data: i32,
+    pub bit_index: u8,
+    pub gain: Gain,
+}
+
+#[derive(Debug)]
+/// This represent one data sample for one period conversion. 
+pub struct DataSample {
+    /// Represent the 24-bit signed (two's complement) data output. 
+    pub data_out: i32,
+    /// Represent the applied gain for next conversion period.     
+    pub next_gain: Gain,
+}
+
+/// The gain act as the 'stop index' for the pulses.
+/// Where:
+/// - Gain(128) => 25 bit pulses.
+/// - Gain(32) => 26 bit pulses.
+/// - Gain(64) => 27 bit pulses.
+#[repr(u8)]
+#[derive(Debug, Clone, Copy)]
+pub enum Gain {
+    Apply128 = 25,
```

Comment:
Move doc comments to here instead. Each public enum variant should have doc comments